[Bug]: Editing Playlists with Android Application Audinaut fails

### What happened?

Adding Songs in Audinaut to an existing Airsonic Advanced Playlist fails with an Error in the Logs of both Airsonic Advanced (see logs below) and Audinaut: 
Failed to update "Playlistname", please try later. Server error: could not execute statement; SQL [n/a]; constraint [null]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement.

Editing the Playlist in that way was working with 11.1.2 (Stable) and the versions before. It is broken after the upgrade to 11.1.3 (Stable).

### Steps to reproduce

1. Use Audinaut (Android - Fdroid Store) with Arisonic advanced 11.1.3.
2. Create a new Playlist with one Song.
3. Try to add another Song to this Playlist

### Version

11.1.x (Edge)

### Version Detail

11.1.3 (Stable)

### Operating System

Linux (RaspberryPi OS)

### Java Version

OpenJDK Runtime Environment (build 17.0.10+7-Debian-1deb11u1)

### Database

Other

### DB Detail

Build-in HSQL

### Configuration paramter

EncryptionKeyPassword=xxx
EncryptionKeySalt=xxx
SettingsChanged=1707217826974
JWTKey=xxx
IndexCreationInterval=7
IndexCreationHour=4
FastCacheEnabled=true
OrganizeByFolderStructure=true
UploadsFolder=%{['USER_MUSIC_FOLDERS'][0]}/Incoming
ExcludePattern=
IgnoreSymLinks=false
IndexString=A B C D E F G H I J K L M N O P Q R S T U V W X-Z(XYZ)
IgnoredArticles=The El La Los Las Le Les Der Die Das
GenreSeparators=;
Shortcuts=New Incoming Podcast
PlaylistFolder=/var/playlists
MusicFileTypes=mp3 ogg oga aac m4a m4b flac wav wma aif aiff ape mpc shn mka opus alm 669 mdl far xm mod fnk imf it liq wow mtm ptm rtm stm s3m ult dmf dbm med okt emod sfx m15 mtn amf gdm stx gmc psm j2b umx amd rad hsc flx gtk mgt mtp wv
VideoFileTypes=flv avi mpg mpeg mp4 m4v mkv mov wmv ogv divx m2ts webm
CoverArtFileTypes2=cover.jpg cover.png cover.gif folder.jpg jpg jpeg gif png
CoverArtSource=TAGFILE
CoverArtConcurrency=4
CoverArtQuality=90
SortAlbumsByYear=true
GettingStartedEnabled=false
WelcomeTitle=Welcome to Airsonic!
WelcomeSubtitle=
WelcomeMessage2=
LoginMessage=
Theme=default
LocaleLanguage=en
LocaleCountry=
LocaleVariant=
spring.liquibase.parameters.mysqlVarcharLimit=384
DbBackupUpdateInterval=720
DbBackupRetentionCount=1
server.servlet.session.timeout=2h
HideIndexedFiles=true
EnableCueIndexing=true

### Proxy Server

none

### client detail

Audinaut (Fdroid Store)

### language

English

### Relevant log output

```console
at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:475) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:344) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:40) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:107) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1407) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:489) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:3303) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:2438) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:449) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:183) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.access$300(JdbcResourceLocalTransactionCoordinatorImpl.java:40) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:281) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:101) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:562) ~[spring-orm-5.3.31.jar!/:5.3.31]
... 137 common frames omitted
Caused by: java.sql.SQLIntegrityConstraintViolationException: integrity constraint violation: NOT NULL check constraint ; SYS_CT_10381 table: PLAYLIST column: NAME
at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.jdbc.JDBCPreparedStatement.fetchResult(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.jdbc.JDBCPreparedStatement.executeUpdate(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-4.0.3.jar!/:na]
at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-4.0.3.jar!/:na]
at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
... 159 common frames omitted
Caused by: org.hsqldb.HsqlException: integrity constraint violation: NOT NULL check constraint ; SYS_CT_10381 table: PLAYLIST column: NAME
at org.hsqldb.error.Error.error(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.Table.enforceRowConstraints(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.Table.generateAndCheckData(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.Table.insertSingleRow(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.StatementDML.update(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.StatementDML.executeUpdateStatement(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.StatementDML.getResult(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.StatementDMQL.execute(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.Session.executeCompiledStatement(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
at org.hsqldb.Session.execute(Unknown Source) ~[hsqldb-2.7.1.jar!/:2.7.1]
... 164 common frames omitted
```

--- a/airsonic-main/pom.xml
+++ b/airsonic-main/pom.xml
@@ -9,7 +9,7 @@
     <parent>
         <groupId>org.airsonic.player</groupId>
         <artifactId>airsonic</artifactId>
-        <version>11.1.3-SNAPSHOT</version>
+        <version>11.1.4-SNAPSHOT</version>
     </parent>
 
     <properties>

--- a/airsonic-main/src/main/java/org/airsonic/player/service/PlaylistService.java
+++ b/airsonic-main/src/main/java/org/airsonic/player/service/PlaylistService.java
@@ -334,7 +334,7 @@ public class PlaylistService {
     @Transactional
     public void updatePlaylist(Integer id, String name, String comment, Boolean shared) {
         playlistRepository.findById(id).ifPresentOrElse(p -> {
-            p.setName(name);
+            if (name != null) p.setName(name);
             if (comment != null) p.setComment(comment);
             if (shared != null) p.setShared(shared);
             p.setChanged(Instant.now());

--- a/airsonic-main/src/test/java/org/airsonic/player/repository/AlbumRepositoryTest.java
+++ b/airsonic-main/src/test/java/org/airsonic/player/repository/AlbumRepositoryTest.java
@@ -38,6 +38,7 @@ import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.data.domain.Sort;
 import org.springframework.data.domain.Sort.Direction;
+import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.transaction.annotation.Transactional;
 
 import java.nio.file.Path;
@@ -68,6 +69,9 @@ public class AlbumRepositoryTest {
     @Autowired
     private MusicFolderRepository musicFolderRepository;
 
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
     @TempDir
     private static Path tempDir;
 
@@ -88,6 +92,7 @@ public class AlbumRepositoryTest {
         testFolders.add(musicFolder2);
         testFolders.add(musicFolder3);
         musicFolderRepository.saveAll(testFolders);
+        jdbcTemplate.execute("DELETE FROM album WHERE present = false");
     }
 
     @AfterEach
@@ -113,7 +118,6 @@ public class AlbumRepositoryTest {
             assertTrue(result.isPresent());
             assertEquals(album, result.get());
             assertFalse(notFound.isPresent());
-
         }
 
         @Test
@@ -145,7 +149,6 @@ public class AlbumRepositoryTest {
         public void testMarkNonPresent() {
 
             // given
-
             long count = albumRepository.findByPresentFalse().size();
             Instant now = Instant.now().minusSeconds(3600);
             Album album1 = new Album("path1", "name1", "artist1", Instant.now(), now, true, testFolders.get(0));

--- a/airsonic-main/src/test/java/org/airsonic/player/repository/ArtistRepositoryTest.java
+++ b/airsonic-main/src/test/java/org/airsonic/player/repository/ArtistRepositoryTest.java
@@ -38,6 +38,7 @@ import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.data.domain.Sort;
 import org.springframework.data.domain.Sort.Direction;
+import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.transaction.annotation.Transactional;
 
 import java.nio.file.Path;
@@ -66,6 +67,9 @@ public class ArtistRepositoryTest {
     @Autowired
     private MusicFolderRepository musicFolderRepository;
 
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
     @TempDir
     private static Path tempDir;
 
@@ -83,6 +87,7 @@ public class ArtistRepositoryTest {
         testFolders.add(musicFolder);
         testFolders.add(musicFolder2);
         musicFolderRepository.saveAll(testFolders);
+        jdbcTemplate.execute("DELETE FROM artist WHERE present = false");
     }
 
     @AfterEach

--- a/airsonic-sonos-api/pom.xml
+++ b/airsonic-sonos-api/pom.xml
@@ -8,7 +8,7 @@
     <parent>
         <groupId>org.airsonic.player</groupId>
         <artifactId>airsonic</artifactId>
-        <version>11.1.3-SNAPSHOT</version>
+        <version>11.1.4-SNAPSHOT</version>
     </parent>
 
     <dependencies>

--- a/install/docker/pom.xml
+++ b/install/docker/pom.xml
@@ -5,7 +5,7 @@
     <parent>
         <relativePath>../../pom.xml</relativePath>
         <artifactId>airsonic</artifactId>
-        <version>11.1.3-SNAPSHOT</version>
+        <version>11.1.4-SNAPSHOT</version>
         <groupId>org.airsonic.player</groupId>
     </parent>
     <packaging>pom</packaging>
@@ -16,7 +16,7 @@
         <dependency>
             <groupId>org.airsonic.player</groupId>
             <artifactId>airsonic-main</artifactId>
-            <version>11.1.3-SNAPSHOT</version>
+            <version>11.1.4-SNAPSHOT</version>
             <type>war</type>
         </dependency>
     </dependencies>

--- a/integration-test/pom.xml
+++ b/integration-test/pom.xml
@@ -8,7 +8,7 @@
     <parent>
         <groupId>org.airsonic.player</groupId>
         <artifactId>airsonic</artifactId>
-        <version>11.1.3-SNAPSHOT</version>
+        <version>11.1.4-SNAPSHOT</version>
     </parent>
 
     <properties>

--- a/pom.xml
+++ b/pom.xml
@@ -5,7 +5,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>org.airsonic.player</groupId>
     <artifactId>airsonic</artifactId>
-    <version>11.1.3-SNAPSHOT</version>
+    <version>11.1.4-SNAPSHOT</version>
     <name>Airsonic</name>
     <packaging>pom</packaging>
     <organization>

--- a/subsonic-rest-api/pom.xml
+++ b/subsonic-rest-api/pom.xml
@@ -8,7 +8,7 @@
     <parent>
         <groupId>org.airsonic.player</groupId>
         <artifactId>airsonic</artifactId>
-        <version>11.1.3-SNAPSHOT</version>
+        <version>11.1.4-SNAPSHOT</version>
     </parent>
 
     <dependencies>

